# Task - 2

# Exercise 7: The Numpy Package: Working with numbers made easy!

1. Import Numpy!

In [ ]:
import numpy as np

2. You have the following stock Prices in EUR Currency : [23.65, 43.21, 73.01, 87.32]
   Create a **Numpy array** and **convert all prices** to USD Currency @ an fx rate of 1.2 (1.2 USD per 1 EUR).


In [ ]:
arr=np.array([23.65,43.21,73.01,87.32])

In [ ]:
usd_price = arr*1.2
print(usd_price)



[ 28.38   51.852  87.612 104.784]


You have the

end-of-2017 stock prices for four companies: [33.65, 63.21, 53.01, 17.32]
end-of-2018 stock prices for the same companies: [36.54, 61.59, 60.31, 10.40]

3. Calculate the **absolute** price increase / decrease in 2018 **for all four companies.** Use **vectorized** numpy code!

In [ ]:
end_of_2017 =np.array([33.65, 63.21, 53.01, 17.32])
end_of_2018=np.array([36.54, 61.59, 60.31, 10.40])
absolute_price=np.abs(end_of_2017 - end_of_2018)
absolute_price

array([2.89, 1.62, 7.3 , 6.92])

4. Calculate the **price return** in 2018 **for all four companies.** Use **vectorized** numpy code!

In [ ]:
price_return=(end_of_2018 - end_of_2017)/end_of_2017
price_return

array([ 0.0858841 , -0.02562886,  0.13770987, -0.39953811])

You have the following list with cashflows (starting today at t0): [-500, -100, 200, 600, 1000, 800]

In [ ]:
#run the cell!
l  = [-500, -100, 200, 600, 1000, 800]

5. Convert l into a **numpy array** and **save** the array in memory! **Select** the **second cashflow** at t1.

In [ ]:
arr=np.array(l)
arr

array([-500, -100,  200,  600, 1000,  800])

In [ ]:
t1=arr[1]
t1

-100

6. **Slice** the array for the **last two cashflows!**

In [ ]:
last_two_cashflows=arr[-2:]
last_two_cashflows

array([1000,  800])

7. **Slice** the array for the cashflows in **t2, t3 and t4.**

In [ ]:
arr[2:5]

array([ 200,  600, 1000])

8. The machine costs 500 USD (Investment in to). Calculate the **Project´s NPV** and evaluate whether ABC should pursue the project.
   ABC´s required rate of return (Cost of Capital) is 7.5% p.a. Use **vectorized** numpy code!

In [ ]:
# run the cell
cf = np.array([-500, 50, 80, 120, 150, 200])
cf

array([-500,   50,   80,  120,  150,  200])

In [ ]:
r=0.075
discount_factors=1/(1+r)**np.arange(len(cf))
discounted_cashflows=cf*discount_factors
npv = np.sum(discounted_cashflows)
if npv > 0:
    decision = "ABC should pursue the project."
elif npv < 0:
    decision = "ABC should not pursue the project."
print("NPV:", npv)
print("Decision:", decision)

NPV: -36.03468883756554
Decision: ABC should not pursue the project.


9. **Slice** the array cf (from question 8) for the positive inflows (from t1) and save the slice in the variable cf_pos.
   Make sure that once you **change the slice cf_pos**, the array cf **shall change as well!**

In [ ]:
cf_pos=[]
for i in cf:
  if i>0:
    cf_pos.append(i)
print(cf_pos)


[50, 80, 120, 150, 200]


In [ ]:
cf_pos[2]=cf_pos[3]
cf_pos


[50, 80, 150, 150, 200]

In [ ]:
cf=cf_pos
cf

[50, 80, 150, 150, 200]

10. Change the first cashflow (t1) of **cf_pos to 70**. Check whether **cf gets mutated as well!**


In [ ]:
cf_pos[1]=70
cf_pos

[50, 70, 150, 150, 200]

In [ ]:
#yes it gets mutated as well.

11. **Restore cf** and redo **Q9 and Q10**. This time make sure that **cf and cf_pos** are independent objects!

In [ ]:
cf


[50, 70, 150, 150, 200]

In [ ]:
cf = np.array([-500, 50, 80, 120, 150, 200])
cf

array([-500,   50,   80,  120,  150,  200])

12. Calculate the total **sum** over all cashflows of cf!

In [ ]:
# run the cell
cf = np.array([-500, 50, 80, 120, 150, 200])
cf

In [ ]:
cf_pos[1]=70
cf_pos

[50, 70, 150, 150, 200]

In [ ]:
cf

array([-500,   50,   80,  120,  150,  200])

13. Return the **highest Value** of cf!

In [ ]:
max(cf)

200

14. Calculate the **cumulative cashflows** for cf!

In [ ]:
cumulative_cashflows=np.cumsum(cf)
cumulative_cashflows

array([-500, -450, -370, -250, -100,  100])

15. Assume the following list: [3, 10, 6, 45, 100]. Calculate the **square root** for all elements in the list.

In [ ]:
l=[3, 10, 6, 45, 100]
l

[3, 10, 6, 45, 100]

In [ ]:
import math
square_roots = []

for num in l:
    square_roots.append(math.sqrt(num))

print(square_roots)

[1.7320508075688772, 3.1622776601683795, 2.449489742783178, 6.708203932499369, 10.0]


Assume the following stock prices:

In [ ]:
# run the cell!
a = np.array([0.34, 43.56, 32.12, 67.43, 0.87, 9.87])
a

array([ 0.34, 43.56, 32.12, 67.43,  0.87,  9.87])

16. **Filter** a for all penny stocks! (price less than 1)

In [ ]:
penny_stocks=[]
for i in a:
  if i<1:
    penny_stocks.append(i)
penny_stocks

[0.34, 0.87]

# Exercise 8: N-Dimensional Numpy Arrays

In [ ]:
# run the cell!
import numpy as np
np.set_printoptions(precision=2, suppress= True)

Let´s assume we have three investment projects with the following cashflow plans:

In [ ]:
# run the cell!
project_1 = [-500, 100, 150, 200, 200, 175, 100, 50]
project_2 = [-750, 100, 125, 150, 175, 200, 250, 0]
project_3 = [-2000, 500, 1000, 2500, 0, 0, 0, 0]

1. Create and save a **2-dimensional Numpy Array** (in the variable projects).

In [ ]:
data=np.array([project_1,project_2,project_3])
data

array([[ -500,   100,   150,   200,   200,   175,   100,    50],
       [ -750,   100,   125,   150,   175,   200,   250,     0],
       [-2000,   500,  1000,  2500,     0,     0,     0,     0]])

2. Check the **shape** of projects. How many **rows/columns** do we have?

In [ ]:
np.shape(data)

(3, 8)

In [ ]:
#3 rows and 8 columns

3. **Slice** projects for **the second project.**

In [ ]:
data[1]



array([-750,  100,  125,  150,  175,  200,  250,    0])

4. Return the **initial Investment** of Project 3.

In [ ]:
for i in data[2]:
  print(i)
  break

-2000


5. Return the **final cashflow** in year 7 for **all projects.**

In [ ]:
final_cashflow_7=data[ : ,7]
final_cashflow_7

array([50,  0,  0])

6. Return the first positive cashflows **(Year 1)** for **Projects 1 and 2.**

In [ ]:
project_1 =np.array( [-500, 100, 150, 200, 200, 175, 100, 50])
project_2 = np.array([-750, 100, 125, 150, 175, 200, 250, 0])
project_3 = np.array([-2000, 500, 1000, 2500, 0, 0, 0, 0])
index_project_1 = np.argmax(project_1 > 0)
index_project_2 = np.argmax(project_2 > 0)
positive_cashflow_project_1 = project_1[index_project_1]
positive_cashflow_project_2 = project_2[index_project_2]
print("First Positive Cashflow for Project 1 (Year 1):", positive_cashflow_project_1)
print("First Positive Cashflow for Project 2 (Year 1):", positive_cashflow_project_2)

First Positive Cashflow for Project 1 (Year 1): 100
First Positive Cashflow for Project 2 (Year 1): 100


7. **Change** the initial investment for **project 3 to -1900.**

In [ ]:
project_3[0]=-1900
project_3

array([-1900,   500,  1000,  2500,     0,     0,     0,    25])

8. Change the **last cashflows** in year 7 for all projects to **25.**

In [ ]:
project_1[7]=25
project_2[7]=25
project_3[7]=25
print(project_1)
print(project_2)
print(project_3)

[-500  100  150  200  200  175  100   25]
[-750  100  125  150  175  200  250   25]
[-1900   500  1000  2500     0     0     0    25]


9. Calculate the total cashflows on each timestamp over all three projects **(column-wise sum).**


In [ ]:
data=np.array([project_1,project_2,project_3])
data
sum=np.sum(data,0)
sum

array([-3150,   700,  1275,  2850,   375,   375,   350,    75])

(results: array([-3150, 700, 1275, 2850, 375, 375, 350, 75]))

10. Calculate **cumulative cashflows** for all three projects.

In [ ]:
for_1=np.cumsum(project_1)
for_2=np.cumsum(project_2)
for_3=np.cumsum(project_3)
print(for_1)
print(for_2)
print(for_3)

[-500 -400 -250  -50  150  325  425  450]
[-750 -650 -525 -375 -200    0  250  275]
[-1900 -1400  -400  2100  2100  2100  2100  2125]


In [ ]:
#the calculations assume that the cash flows are provided in chronological order, and the initial cash flow for each project is considered in the cumulative calculation.

11. Convert **all projects** to EUR currency **(multiply with 0.9).**

In [ ]:
eur_1=0.9*project_1
eur_2=0.9*project_2
eur_3=0.9*project_3
print(eur_1)
print(eur_2)
print(eur_3)

[-450.    90.   135.   180.   180.   157.5   90.    22.5]
[-675.    90.   112.5  135.   157.5  180.   225.    22.5]
[-1710.    450.    900.   2250.      0.      0.      0.     22.5]


# Exercise 9: Pandas - Importing and Inspecting your own Dataset

Before we can work with the Pandas Library, we need to import Pandas (by convention as pd).

1. **Import** Pandas. Fill in the gaps!

In [ ]:
import pandas as pd
('pip install pandas')

'pip install pandas'

In [102]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Now we want to import the **Cars Dataset** that is stored in the CSV-File **cars.csv.**

In [107]:
df = pd.read_csv('/content/drive/MyDrive/cars.csv.csv')
df

mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin                       name  
0            70     usa  chevrolet chevelle malibu  
1            70     usa          buick skylark 320  
2            70     usa         plymouth satellite  
3            70     usa              amc rebel sst  
4            70     usa                ford torino  
..          ...     ...                        ...  
393          82     usa            ford mustang gl  
394          82  europe                  vw pickup  
395          82     usa              dodge rampage  
396          82     usa                ford ranger  
397          82     usa             chevrolet s-10  

[398 rows x 9 columns]

**Some more details on the Columns:**

**mpg:** Measures fuel efficiency in terms of miles per gallon. Higher mpg means higher efficiency

**cylinders:** Number of cylinders in the car

**displacement:** Technical feature (engine)

**horsepower:** Unit of measurement for engine power

**weight:** weight in lbs

**acceleration:** time to accelerate from 0 to 60 mph (sec.)

**model_year:** self explanatory

**origin:** origin of the manufacturer/car

**name:** model name

3a. **Inspect** the DataFrame cars by "printing" cars!

In [110]:
print(df)

      mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin                       name  
0            70     usa  chevrolet chevelle malibu  
1            70     usa      

3b. **Inspect** the first 10 Rows of the DataFrame cars! Fill in the gaps! What is the name of the very first car in the first row (index label/pos = 0)?

In [114]:
print(df.head(10))

# Access the car name in the first row
first_car_name = df.iloc[0]['name']
print("Name of the first car:", first_car_name)

    mpg  cylinders  displacement  horsepower  weight  acceleration  \
0  18.0          8         307.0       130.0    3504          12.0   
1  15.0          8         350.0       165.0    3693          11.5   
2  18.0          8         318.0       150.0    3436          11.0   
3  16.0          8         304.0       150.0    3433          12.0   
4  17.0          8         302.0       140.0    3449          10.5   
5  15.0          8         429.0       198.0    4341          10.0   
6  14.0          8         454.0       220.0    4354           9.0   
7  14.0          8         440.0       215.0    4312           8.5   
8  14.0          8         455.0       225.0    4425          10.0   
9  15.0          8         390.0       190.0    3850           8.5   

   model_year origin                       name  
0          70    usa  chevrolet chevelle malibu  
1          70    usa          buick skylark 320  
2          70    usa         plymouth satellite  
3          70    usa         

It´s a ... Chevrolet Chevelle Malibu!

3c. Also inspect the **last 5 Rows! How many cars/rows** do we have in the Dataset (mind the range index starting from 0!)?

In [115]:
print(df.tail(5))

      mpg  cylinders  displacement  horsepower  weight  acceleration  \
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin             name  
393          82     usa  ford mustang gl  
394          82  europe        vw pickup  
395          82     usa    dodge rampage  
396          82     usa      ford ranger  
397          82     usa   chevrolet s-10  


In [117]:
df.shape[0]


398

We have ... 398 Cars!

4. **Inspect** the entire DataFrame **(all rows)! Fill in the gaps!**


In [119]:
for index, row in df.iterrows():
    print("Row index:", index)
    print("Row data:")
    print(row)
    print("\n")


Streaming output truncated to the last 5000 lines.


Row index: 41
Row data:
mpg                          14.0
cylinders                       8
displacement                318.0
horsepower                  150.0
weight                       4096
acceleration                 13.0
model_year                     71
origin                        usa
name            plymouth fury iii
Name: 41, dtype: object


Row index: 42
Row data:
mpg                          12.0
cylinders                       8
displacement                383.0
horsepower                  180.0
weight                       4955
acceleration                 11.5
model_year                     71
origin                        usa
name            dodge monaco (sw)
Name: 42, dtype: object


Row index: 43
Row data:
mpg                                 13.0
cylinders                              8
displacement                       400.0
horsepower                         170.0
weight                              4746
accele

5. Get some **meta information** on our DataFrame! In which **Column** do we have **missing/NaN Values?**

In [120]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    392 non-null    float64
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model_year    398 non-null    int64  
 7   origin        398 non-null    object 
 8   name          398 non-null    object 
dtypes: float64(4), int64(3), object(2)
memory usage: 28.1+ KB
None


In [ ]:
#in column 3 we have missing values because it is less than total entries.

We have 6 missing/NaN Values in the column ... horsepower!

6. Let´s get some **summary statistics** on our DataFrame! What is the **maximum** value in the **column horsepower?**

In [121]:
summary_stats = df.describe()


max_horsepower = summary_stats.loc["max", "horsepower"]
max_horsepower

230.0

The maximum value in the column horsepower is ... 230!

# Exercise 10: Indexing and Slicing DataFrames

7. **Import Pandas (as pd)**,

    **Import** the Cars Dataset **(cars.csv)** and save the DataFrame in the variable cars and
    
    **Inspect/Print cars!**

In [122]:
#Import libraries
import pandas as pd

In [123]:
#import dataset
cars = pd.read_csv('/content/drive/MyDrive/cars.csv.csv')
cars

mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin                       name  
0            70     usa  chevrolet chevelle malibu  
1            70     usa          buick skylark 320  
2            70     usa         plymouth satellite  
3            70     usa              amc rebel sst  
4            70     usa                ford torino  
..          ...     ...                        ...  
393          82     usa            ford mustang gl  
394          82  europe                  vw pickup  
395          82     usa              dodge rampage  
396          82     usa                ford ranger  
397          82     usa             chevrolet s-10  

[398 rows x 9 columns]

In [124]:
#inspect
print(cars)

      mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin                       name  
0            70     usa  chevrolet chevelle malibu  
1            70     usa      

8. Select the column **"name"** with the **attribute** (dot) notation! What is the **name** of the car with **index label 393?**

In [126]:
name_column=cars.name
car_name=name_column.loc[393]
car_name

'ford mustang gl'

The name is ... ford mustang gl!

9. Apply also the **second alternative** to select the column **"name"!**

In [127]:
name_column=cars['name']
car_name=name_column.loc[393]
car_name

'ford mustang gl'

10. **Select** the car/row at **index position 100** with the iloc operator (all columns)! How many **cylinders** does the car have?

In [186]:
car_at_100 = cars.iloc[100]

num_cylinders = car_at_100['cylinders']
num_cylinders


6

11. Select the car/row at **index position 200** with the **iloc** operator (and only the column "name")!
    What is the **name** of the car?

In [136]:
car_at_100 = cars.iloc[100, 8]
car_at_100

'ford maverick'

12. Select the **last 10 rows** and the columns **mpg, horsepower, origin** and name with the **iloc** operator! Fill in the Gaps!
    What is the mpg of the vw pickup?

In [142]:
subset = cars.iloc[-10:, [0, 3, 6, 8]]


vw_pickup_row = subset[subset['name'] == 'vw pickup']


vw_pickup_mpg = vw_pickup_row['mpg'].values[0]

print("MPG of the VW pickup:", vw_pickup_mpg)
subset

MPG of the VW pickup: 44.0


mpg  horsepower  model_year                        name
388  26.0        92.0          82  chrysler lebaron medallion
389  22.0       112.0          82              ford granada l
390  32.0        96.0          82            toyota celica gt
391  36.0        84.0          82           dodge charger 2.2
392  27.0        90.0          82            chevrolet camaro
393  27.0        86.0          82             ford mustang gl
394  44.0        52.0          82                   vw pickup
395  32.0        84.0          82               dodge rampage
396  28.0        79.0          82                 ford ranger
397  31.0        82.0          82              chevrolet s-10

13. **Import** the Cars Dataset one more time (save in the variable cars) and set the column "name" as the index!
    **Fill in the Gaps!**

In [148]:
cars = pd.read_csv('/content/drive/MyDrive/cars.csv.csv', index_col = 'name')
cars

mpg  cylinders  displacement  horsepower  weight  \
name                                                                           
chevrolet chevelle malibu  18.0          8         307.0       130.0    3504   
buick skylark 320          15.0          8         350.0       165.0    3693   
plymouth satellite         18.0          8         318.0       150.0    3436   
amc rebel sst              16.0          8         304.0       150.0    3433   
ford torino                17.0          8         302.0       140.0    3449   
...                         ...        ...           ...         ...     ...   
ford mustang gl            27.0          4         140.0        86.0    2790   
vw pickup                  44.0          4          97.0        52.0    2130   
dodge rampage              32.0          4         135.0        84.0    2295   
ford ranger                28.0          4         120.0        79.0    2625   
chevrolet s-10             31.0          4         119.0        82.0    2720   

                           acceleration  model_year  origin  
name                                                         
chevrolet chevelle malibu          12.0          70     usa  
buick skylark 320                  11.5          70     usa  
plymouth satellite                 11.0          70     usa  
amc rebel sst                      12.0          70     usa  
ford torino                        10.5          70     usa  
...                                 ...         ...     ...  
ford mustang gl                    15.6          82     usa  
vw pickup                          24.6          82  europe  
dodge rampage                      11.6          82     usa  
ford ranger                        18.6          82     usa  
chevrolet s-10                     19.4          82     usa  

[398 rows x 8 columns]

In [149]:
#run the cell and inspect!
cars

mpg  cylinders  displacement  horsepower  weight  \
name                                                                           
chevrolet chevelle malibu  18.0          8         307.0       130.0    3504   
buick skylark 320          15.0          8         350.0       165.0    3693   
plymouth satellite         18.0          8         318.0       150.0    3436   
amc rebel sst              16.0          8         304.0       150.0    3433   
ford torino                17.0          8         302.0       140.0    3449   
...                         ...        ...           ...         ...     ...   
ford mustang gl            27.0          4         140.0        86.0    2790   
vw pickup                  44.0          4          97.0        52.0    2130   
dodge rampage              32.0          4         135.0        84.0    2295   
ford ranger                28.0          4         120.0        79.0    2625   
chevrolet s-10             31.0          4         119.0        82.0    2720   

                           acceleration  model_year  origin  
name                                                         
chevrolet chevelle malibu          12.0          70     usa  
buick skylark 320                  11.5          70     usa  
plymouth satellite                 11.0          70     usa  
amc rebel sst                      12.0          70     usa  
ford torino                        10.5          70     usa  
...                                 ...         ...     ...  
ford mustang gl                    15.6          82     usa  
vw pickup                          24.6          82  europe  
dodge rampage                      11.6          82     usa  
ford ranger                        18.6          82     usa  
chevrolet s-10                     19.4          82     usa  

[398 rows x 8 columns]

In [150]:
#run the cell and inspect!
cars.info()

<class 'pandas.core.frame.DataFrame'>
Index: 398 entries, chevrolet chevelle malibu to chevrolet s-10
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    392 non-null    float64
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model_year    398 non-null    int64  
 7   origin        398 non-null    object 
dtypes: float64(4), int64(3), object(1)
memory usage: 28.0+ KB


14. **Inspect** the new Index of the DataFrame by calling the index attribute!

In [153]:
cars.index

Index(['chevrolet chevelle malibu', 'buick skylark 320', 'plymouth satellite',
       'amc rebel sst', 'ford torino', 'ford galaxie 500', 'chevrolet impala',
       'plymouth fury iii', 'pontiac catalina', 'amc ambassador dpl',
       ...
       'chrysler lebaron medallion', 'ford granada l', 'toyota celica gt',
       'dodge charger 2.2', 'chevrolet camaro', 'ford mustang gl', 'vw pickup',
       'dodge rampage', 'ford ranger', 'chevrolet s-10'],
      dtype='object', name='name', length=398)

15. Select the **row** with the index label **"ford ranger"** (use the loc operator)! What is the **weight?**

In [166]:
car_name=cars.loc['ford ranger']
weight = car_name['weight']
print(car_name)
print(weight)

mpg              28.0
cylinders           4
displacement    120.0
horsepower       79.0
weight           2625
acceleration     18.6
model_year         82
origin            usa
Name: ford ranger, dtype: object
2625


16. Select the **row** with the index label **"ford torino"** and only the columns **"model_year" and "origin"!**
    What´s the model year?

In [170]:
car_name=cars.loc[['ford torino'],['model_year','origin']]
model_year=car_name['model_year']
model_year

name
ford torino    70
Name: model_year, dtype: int64

17. Select the columns **"mpg" and "weight"** (all rows) with the **loc operator!**

In [185]:
car = cars.loc[ :,['mpg', 'weight']]
car

mpg  weight
name                                   
chevrolet chevelle malibu  18.0    3504
buick skylark 320          15.0    3693
plymouth satellite         18.0    3436
amc rebel sst              16.0    3433
ford torino                17.0    3449
...                         ...     ...
ford mustang gl            27.0    2790
vw pickup                  44.0    2130
dodge rampage              32.0    2295
ford ranger                28.0    2625
chevrolet s-10             31.0    2720

[398 rows x 2 columns]

In [184]:
cars['mpg']

name
chevrolet chevelle malibu    18.0
buick skylark 320            15.0
plymouth satellite           18.0
amc rebel sst                16.0
ford torino                  17.0
                             ... 
ford mustang gl              27.0
vw pickup                    44.0
dodge rampage                32.0
ford ranger                  28.0
chevrolet s-10               31.0
Name: mpg, Length: 398, dtype: float64